### PR TITLE
Fixed throw an exception in the soft button manager if two buttons have the same name

### LINKED ISSUE #1897: Throw an exception in the soft button manager if two buttons have the same name
### Bug Report
Currently in the `SDLSoftButtonManager`, if two buttons have the same name, an error is logged. We should instead throw an exception since the soft buttons will be removed from the display because the soft button property is set to an empty array.

##### Reproduction Steps
1. Create two soft button objects with the same name. Send the soft buttons using the screen manager.

##### Expected Behavior
An exception should be thrown.

##### Observed Behavior
An error is logged and the `softButtonObjects` property is set to an empty array. 

##### OS & Version Information
* iOS Version: n/a
* SDL iOS Version: 7.0
* Testing Against: n/a

### PR DESCRIPTION
**Pull request by FrankElias77**
Fixes #1897 

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
Unit test handled to expect exception in case of softButtons do not have unique titles.

#### Core Tests
N/A

Core version / branch / commit hash / module tested against: 7.1
HMI name / version / branch / commit hash / module tested against: N/A

### Summary
Exception is thrown in case multiple softButtons are being and do not have unique titles.

##### Breaking Changes
* N/A

##### Enhancements
* N/A

##### Bug Fixes
* N/A

### Tasks Remaining:
N/A

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/private/SDLError.h [hunk 1]
```diff
@@ -95,6 +95,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface NSException (SDLExceptions)
 
++ (NSException *)sdl_duplicateSoftButtonsTitleException;
```
Comment (joeljfischer) [inline SmartDeviceLink/private/SDLError.h:98]:
```suggestion
+ (NSException *)sdl_duplicateSoftButtonsNameException;
```

File: SmartDeviceLink/private/SDLError.m [hunk 2]
```diff
@@ -407,6 +407,10 @@ + (NSError *)sdl_audioStreamManager_notConnected {
 
 @implementation NSException (SDLExceptions)
 
++ (NSException *)sdl_duplicateSoftButtonsTitleException {
+    return [NSException exceptionWithName:@"InvalidSoftButtonsInitialization" reason:@"Attempting to create soft buttons with the same title" userInfo:nil];
```
Comment (joeljfischer) [inline SmartDeviceLink/private/SDLError.m:411]:
```suggestion
+ (NSException *)sdl_duplicateSoftButtonsNameException {
    return [NSException exceptionWithName:@"InvalidSoftButtonsInitialization" reason:@"Attempting to create soft buttons with the same name" userInfo:nil];
```

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLRegisterAppInterfaceSpec.m [hunk 3]
```diff
@@ -42,7 +42,7 @@
     __block SDLDeviceInfo *info = nil;
     __block SDLAppInfo *appInfo = nil;
     __block SDLTemplateColorScheme *colorScheme = nil;
-    __block SDLMsgVersion * currentSDLMsgVersion = [[SDLMsgVersion alloc] initWithMajorVersion:7 minorVersion:0 patchVersion:0];
+    __block SDLMsgVersion * currentSDLMsgVersion = [[SDLMsgVersion alloc] initWithMajorVersion:7 minorVersion:1 patchVersion:0];
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLRegisterAppInterfaceSpec.m:45]: Why was this change necessary?

**Review by joeljfischer — CHANGES_REQUESTED** [review]

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLRegisterAppInterfaceSpec.m (hunk 3 above)
Comment (FrankElias77) [reply SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLRegisterAppInterfaceSpec.m:45]: @joeljfischer The value for SDLMaxProxyRPCVersion was already set to 7.1.0, but was not set in the spec file which resulted in 3 failing tests

**Review by joeljfischer — APPROVED** [review]

**Pull request merged**